Module5_Vue/src/components/Todo/Todo.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <div class="hello">
+    <h1>{{ msg }}</h1>
+    <h2>Todo's</h2>
+    <input
+      v-model="newTodoVal"
+      v-on:keyup.enter='newTodo(newTodoVal)'
+      placeholder='Add new todo'>
+    <input
+      v-model='filteredVal'
+      placeholder='Enter filtered value'
+    >
+    <ul>
+      <li
+          v-for='item in unarchived(items)' :key='item.id'
+          class='todo' v-bind:class='item.done ? "done" : "inprogress"'
+          v-on:click="todoItemClick(item)"
+        >
+        <button
+          v-on:click.stop='edit(item)'
+          class='button edit'
+        > Edit </button>
+        <button
+          v-on:click.stop='remove(item)'
+          class='button remove'
+        > - </button>
+        <button
+          v-on:click.stop='archive(item)'
+          class='button archive'
+        > Archive </button>
+        <span> {{ item.id }}. </span>
+        <span> {{item.description}} </span>
+        <span class='mark done-mark' v-if='item.done'> Done! </span>
+        <span class='mark done-mark' v-if='!item.done'> In progress </span>
+      </li>
+    </ul>
+    <a v-on:click.stop="goTo('Archive')">Go to archive</a>
+  </div>
+</template>
+
+<script src='../../store/external.store.js'></script>
+<script>
+  export default {
+    name: 'Todo',
+    data() {
+      return {
+        msg: 'Welcome to Todo\'s',
+        items: this.$store.state.todos,
+        newTodoVal: '',
+        filteredVal: '',
+      };
+    },
+    mounted() {
+      this.$getExternalTodos()
+        .then(data => this.$store.commit('INIT_TODOS', data));
```

Comment:
I think for 'INIT_TODOS' action we should use the same variable here and in store.
It can be like Action object with all actions('INIT_TODOS', 'TODOS_ARCHIVE'...)

Reply:
fixed